jstests/aggregation/bugs/server32300.js

```diff
@@ -0,0 +1,19 @@
+// Core ServerSERVER-32300 Aggregation shell helper should not modify options document
+
+load("src/mongo/shell/explainable.js")
+
+let collation = {collation: {locale: "fr", backwards: true}};
+
+// Computes correct result of the aggragation
+// Computes an empty array
+jsTestLog("desired output: " + tojson(db.collection.aggregate([], collation).toArray()));
```

Comment:
Instead of just logging here, we should make assertions on the behavior. In this case, I would recommend inserting some data into a collection, then asserting that the results of an aggregation are the same before and after an explain. Something like:
```
const coll = db.explain_options;
coll.drop();

for (let i = 0; i < 10; ++i) {
    assert.writeOK(coll.insert({_id: i}));
}

const firstResults = coll.aggregate([{$sort: {_id: 1}}], collation).toArray();
// Here we issue an explain in order to verify that 'collation' is not modified to include the explain flag. We do not care about the result of the explain.
assert.commandWorked(coll.explain().aggregate([], collation));
const secondResults = coll.aggregate([{$sort: {_id: 1}}], collation).toArray();
assert.eq(firstResults, secondResults);
```